tensordict/dynamic_shaped_tensor.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
```

Comment:
Should we move this to `tensordict.prototype`?

Reply:
That's fair. I guess we need to hold ourselves to moving stuff out once we're relatively confident the API is stable and there's no major bugs. I think there is some value in having the prototype submodule so that we can merge and try out new features while indicating to the user that they might be brittle.

Maybe tensorclass' time has come!